src/apps/service/apps/jobs/submissions/async.clj

```diff
@@ -0,0 +1,118 @@
+(ns apps.service.apps.jobs.submissions.async
+  (:use [clojure-commons.core :only [remove-nil-values]]
+        [slingshot.slingshot :only [try+ throw+]])
+  (:require [clojure.tools.logging :as log]
+            [clojure-commons.file-utils :as ft]
+            [apps.clients.data-info :as data-info]
+            [apps.clients.notifications :as notifications]
+            [apps.persistence.jobs :as jp]
+            [apps.service.apps.job-listings :as job-listings]
+            [apps.service.apps.jobs.submissions.submit :as submit]
+            [apps.util.config :as config]))
+
+(defn- max-batch-paths-exceeded
+  [max-paths first-list-path first-list-count]
+  (throw+
+    {:type       :clojure-commons.exception/illegal-argument
+     :error      (str "The HT Analysis Path List exceeds the maximum of "
+                      max-paths
+                      " allowed paths.")
+     :path       first-list-path
+     :path-count first-list-count}))
+
+(defn- validate-path-lists
+  [path-lists]
+  (let [[first-list-path first-list] (first path-lists)
+        first-list-count             (count first-list)]
+    (when (> first-list-count (config/path-list-max-paths))
+      (max-batch-paths-exceeded (config/path-list-max-paths) first-list-path first-list-count))
+    (when-not (every? (comp (partial = first-list-count) count second) path-lists)
+      (throw+ {:type  :clojure-commons.exception/illegal-argument
+               :error "All HT Analysis Path Lists must have the same number of paths."}))
+    path-lists))
+
+(defn- get-path-list-contents
+  [user path]
+  (try+
+    (when (seq path) (data-info/get-path-list-contents user path))
+    (catch Object _
+      (log/error (:throwable &throw-context)
+                 "job submission failed: Could not get file contents of path list input.")
```

Comment:
Will the throwable contain the path which failed to be loaded, or would it make sense to put the path which failed into the message?